### PR TITLE
Updated documentation to include trace-inputlocator

### PR DESCRIPTION
**Pull request by Jack-Hawthorne**
has some inclusions once the glossary PR is finished

### CODE REVIEW

**Comment by daren-thomas** [comment]
Can one of the admins verify this patch?

File: docs/how-trace-inputlocator-works.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    - 'created_by' : list of scripts which are responsible for writing the file (empty list if input file)
+    - 'used_by' : list of scripts which read the file (empty list if no dependencies exist)
+    - 'file_path' : the absolute path of the file
+    - 'file_type' : the file extension (e.g. 'xls' or 'dbf')
+    - 'schema' :
+        - variable or sheet name (for xls files)
+            - 'sample_data' : the last non-null entry of the file
+            - 'types_found' : list of python types encountered when reading the file, including None types
+
+Note! The first time metadata is generated, the order of the scripts is important as the 'created_by' list is updated each time a script is run. For example,
```
Comment (daren-thomas) [inline docs/how-trace-inputlocator-works.rst]: I'm surprised by this. Can we meet tomorrow to discuss this?

Comment (Jack-Hawthorne) [reply docs/how-trace-inputlocator-works.rst]: what is surprising about it? if the data is already there but the trace-inputlocator has not been run for a dependency, then how is it supposed to know which script created the data?

Comment (daren-thomas) [reply docs/how-trace-inputlocator-works.rst]: well... the Note! sounds like it's a problem, but the explanation suggests it just works... and would just update the `created_by` when it get's there?

**Pull request merged**